packages/@ember/-internals/glimmer/index.ts

```diff
@@ -142,6 +142,157 @@
     First name: <input type="text" />
   </label>
   ```
+
+  `yield` can also be used with the `hash` helper:
+
+  ```app/templates/application.hbs
+  <DateRanges @value={{@model.date}} as |range|>
+    Start date: {{range.start}}
+    End date: {{range.end}}
+  </DateRanges>
+  ```
+
+  ```app/components/date-ranges.hbs
+  <div>
+    {{yield (hash start=@value.start end=@value.end)}}
+  </div>
+  ```
+
+  Result:
+
+  ```html
+  <div>
+    Start Date: July 1st
```

Comment:
```suggestion
    Start date: July 1st
```

Nitpick, but the initial template uses a lowercase `d`. Same for `End date`.